### PR TITLE
Fix for /tmp permissions issue

### PR DESCRIPTION
**Pull request by beholders-eye**
Also adds more error checks

Tries to fix other permission issues when the container run creates
artifacts, by running inside the container as the current user

Rust's CARGO_HOME/RUSTUP_HOME should be shown to the user

### CODE REVIEW

**Comment by beholders-eye** [comment]
Just to add some more context.

When a YAML has `apt-get ...` apt libraries tries to create `/tmp/apt-*` index files and fails, because the CLI is injecting its tools into `/tmp`, I changed it to use `/var/tmp/yb-inject` instead. So Narwhal's `archiveFile` function won't try to change `/tmp` permissions bits to `0750` instead of `1777`

**Comment by zombiezen** [comment]
Is this ready for review? It's still in draft.

**Comment by beholders-eye** [comment]
It was, then I found some issues testing it a bit more. I'm addressing those.

**Comment by beholders-eye** [comment]
The last commit, added yesterday, has a lot more than just fixes for `/tmp`. It adds fixed up commands to prepare a build container to run as normal user, using the same UID/GID that of the current local user (still fixed to Linux only). PTAL @jamesnaftel, @zombiezen or @glacials

**Comment by beholders-eye** [comment]
Dependent on `narwhal-v0.5.3`. PTAL, anyone :wink: :waffle:

File: .yourbase.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ build_targets:
       - bash -c "go build -ldflags \"-X 'main.date=$(date)'\""
 
   - name: integration
+    host_only: true
```
Comment (beholders-eye) [inline .yourbase.yml:13]: I'll work on a way to inject `yb` itself in the container, so this would work

**Comment by zombiezen** [comment]
There is a lot going on in this PR. Can you break this down into smaller PRs for the individual fixes? That would help me review.

**Pull request closed without merging**